common4j/src/main/com/microsoft/identity/common/java/controllers/BaseController.java

```diff
@@ -262,9 +265,17 @@ protected final AuthorizationRequest.Builder initializeAuthorizationRequestBuild
             Logger.error(TAG, "correlation id from diagnostic context is not a UUID", ex);
         }
 
-        builder.setClientId(parameters.getClientId())
-                .setRedirectUri(parameters.getRedirectUri());
 
+        if (parameters.hasNestedAppParameters()) {
```

Comment:
Does NAA require PRT?

I'm wondering if there should be a subclass specifically for NAA.

Reply:
I tried adding a subclass for NAA. But it needs a lot of refactoring of classes than I expected. I added details on what is blocking me. I will leave this as it is for now as the changes under the if block for NAA are very small. I will take this up refactoring work as a different task parallelly so that the review is blocked.

Reply:
To your first question, yes nested app tokens are also acquired using PRT, NAA work is particularly dependent on PRTV3. Hence you would see changes in this PR are dependent on PRTV3 flight